utils_cv/segmentation/model.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+from functools import partial
+from pathlib import Path
+from typing import List, Union
+
+from fastai.basic_data import DeviceDataLoader
+from fastai.basic_train import Learner
+import numpy as np
+import PIL
+from sklearn.metrics import confusion_matrix as sk_confusion_matrix
+
+from .dataset import load_im
+
+
+# Ignore pixels marked as void. That could be pixels which are hard to annotate and hence should not influence training.
+def _objective_fct_partial(void_id, input, target):
+    """ Helper function to compute the ratio of correctly classified pixels. """
+    target = target.squeeze(1)
+    if void_id:
+        mask = target != void_id
+        ratio_correct = (
+            (input.argmax(dim=1)[mask] == target[mask]).float().mean()
+        )
+    else:
+        ratio_correct = (input.argmax(dim=1) == target).float().mean()
+
+    return ratio_correct
+
+
+def get_objective_fct(classes: List[str]):
+    """ Returns objective function for model training, defined as ratio of correctly classified pixels.
+
+    Args:
+        classes: list of class names
+
+    Return:
+        Objective function.
+    """
+    class2id = {v: k for k, v in enumerate(classes)}
+    if "void" in class2id:
+        void_id = class2id["void"]
+    else:
+        void_id = None
+
+    return partial(_objective_fct_partial, void_id)
+
+
+def predict(
```

Comment:
should this function exist? If feels odd to have a prediction wrapper around fastai's predict function. If the thresh error gets fixed in fastai, then the function is redundant.

Also turning the masks and scores into np arrays can be done inside the plot segmentation function instead (since that is the main function that uses the output of the masks/scores)

Reply:
Good point. Is it OK though if we leave it in, since fastai's predict() seams to have a bug with the threshold (plus this function also loads an image if needed similar to the predict() in object detection.
I am mapping it to a numpy array so that this line works: mask[max_scores <= thres] = 0. Both are 2D matrices, so alternative to numpy could be to return it as a tensor (which I don't think would be better)

Reply:
yeah makes sense